WooCommerce/src/main/kotlin/com/woocommerce/android/util/ApplicationEdgeToEdgeEnabler.kt

```
@@ -9,7 +9,7 @@ import javax.inject.Inject
 
 class ApplicationEdgeToEdgeEnabler @Inject constructor() : Application.ActivityLifecycleCallbacks {
     override fun onActivityCreated(activity: Activity, bindle: Bundle?) {
-        (activity as ComponentActivity).enableEdgeToEdge()
+        (activity as? ComponentActivity)?.enableEdgeToEdge()
```

I think it's kinda dangerous to do this way. We just won't know if some activity didn't apply edge to edge which can end up in different behavior on 35 and below

I am wondering if we should force edge-to-edge only for activities with our package and do not do anything for the rest?

---

> We just won't know if some activity didn't apply edge to edge, which can end up in different behavior on 35 and below

Good point @kidinov.

I refactored the code to:

1. Enable edge-to-edge on supported activities (we have a list of unsupported edge-to-edge activities where we can add exceptional cases)
2. Fail fast when debugging so we can be aware of activities not supporting edge-to-edge
3. Log activities that do not support edge-to-edge when the app is not in debug mode, allowing to continue using the app if this case is reached.

Let me know WDYT?